applications/crossbar/src/modules/cb_modules_util.erl

```diff
@@ -572,27 +572,27 @@ remove_plaintext_password(Context) ->
 %%--------------------------------------------------------------------
 -spec validate_number_ownership(ne_binaries(), cb_context:context()) ->
                                        cb_context:context().
--spec validate_number_ownership([{knm_numbers:num(), knm_numbers:ko()}], ne_binaries(), cb_context:context()) ->
-                                       cb_context:context().
 validate_number_ownership(Numbers, Context) ->
     Options = [{'auth_by', cb_context:auth_account_id(Context)}],
     #{ko := KOs} = knm_numbers:get(Numbers, Options),
-    validate_number_ownership(maps:to_list(KOs), [], Context).
-
-validate_number_ownership([], [], Context) -> Context;
-validate_number_ownership([], Unauthorized, Context) ->
-    Prefix = <<"unauthorized to use ">>,
-    Numbers = kz_binary:join(Unauthorized, <<", ">>),
-    Message = <<Prefix/binary, Numbers/binary>>,
-    cb_context:add_system_error(403, 'forbidden', Message, Context);
-validate_number_ownership([{_, Reason}|KOs], Unauthorized, Context) when is_atom(Reason) ->
+    case maps:fold(fun validate_number_ownership_fold/3, [], KOs) of
+        [] -> Context;
+        Unauthorized ->
+            Prefix = <<"unauthorized to use ">>,
+            NumbersStr = kz_binary:join(Unauthorized, <<", ">>),
+            Message = <<Prefix/binary, NumbersStr/binary>>,
+            cb_context:add_system_error(403, 'forbidden', Message, Context)
+    end.
+
+-spec validate_number_ownership_fold(knm_numbers:num(), knm_numbers:ko(), ne_binaries()) ->
+                                            ne_binaries().
+validate_number_ownership_fold(_, Reason, Unauthorized) when is_atom(Reason) ->
     %% Ignoring atom reasons, i.e. 'not_found' or 'not_reconcilable'
-    validate_number_ownership(KOs, Unauthorized, Context);
-validate_number_ownership([{Number, ReasonJObj}|KOs], Unauthorized, Context) ->
+    Unauthorized;
+validate_number_ownership_fold(Number, ReasonJObj, Unauthorized) ->
     case knm_errors:error(ReasonJObj) of
```

Comment:
Oh do note though that if a number is not reconcilable it fails with the atom `not_reconcilable` instead of a knm_errors JSON object. So you should rewrite the case to something like `case not is_atom(Reason) andalso knm_errors:error(Error) =:= <<"forbidden">> of true -> ...`